theHarvester/discovery/dnssearch.py

```diff
@@ -40,3 +59,191 @@ def process(self):
                 # print(' : ' + host.split(':')[1])
                 results.append(host)
         return results
+
+#####################################################################
+# DNS REVERSE
+#####################################################################
+
+
+IP_REGEX = r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}'
+PORT_REGEX = r'\d{1,5}'
+NETMASK_REGEX = r'\d{1,2}|' + IP_REGEX
+NETWORK_REGEX = r'\b({})(?:\:({}))?(?:\/({}))?\b'.format(
+    IP_REGEX,
+    PORT_REGEX,
+    NETMASK_REGEX)
+
+
+def serialize_ip_range(
+        ip: str,
+        netmask: str = '24') -> str:
+    """
+    Serialize a network range in a constant format, 'x.x.x.x/y'.
+
+    Parameters
+    ----------
+    ip: str.
+        A serialized ip in the format 'x.x.x.x'.
+        Extra information like port (':z') or subnet ('/n')
+        will be ignored.
+    netmask: str.
+        The subnet subdivision, represented by a 2 digit netmask.
+
+    Returns
+    -------
+    out: str.
+        The network OSI address, like '192.168.0.0/24'.
+    """
+    __ip_matches = re.search(NETWORK_REGEX, ip, re.IGNORECASE)
+    __ip = __ip_matches.group(1)
+    __netmask = netmask if netmask else __ip_matches.group(3)
+    if __ip and __netmask:
+        return str(IPv4Network('{}/{}'.format(__ip, __netmask), strict=False))
+    elif __ip:
+        return str(IPv4Network('{}/{}'.format(__ip, '24'), strict=False))
+    else:   # invalid input ip
+        return ''
+
+
+def list_ips_in_network_range(
+        iprange: str) -> List[str]:
+    """
+    List all the IPs in the range.
+
+    Parameters
+    ----------
+    iprange: str.
+        A serialized ip range, like '1.2.3.0/24'.
+        The last digit can be set to anything, it will be ignored.
+
+    Returns
+    -------
+    out: list.
+        The list of IPs in the range.
+    """
+    try:
+        __network = IPv4Network(iprange, strict=False)
```

Comment:
Any reason why strict is set to False?

Reply:
yep: the search engines collects ips like '178.128.43.171' but the 4th number must be set to 0 when creating an IPv4Network object.

so instead of manually replacing the "171" by "0", I ask IPv4Network to be lenient and still create the network "178.128.43.0/24" itself